hypervisor/src/mshv/mod.rs

```diff
@@ -1145,7 +1127,64 @@ impl cpu::Vcpu for MshvVcpu {
                                     // Clear the SW_EXIT_INFO1 register to indicate no error
                                     self.clear_swexit_info1(ghcb_gpa)?;
                                 }
-                                SVM_EXITCODE_SNP_GUEST_REQUEST => {
+                                SVM_EXITCODE_SNP_GUEST_REQUEST
+                                | SVM_EXITCODE_SNP_EXTENDED_GUEST_REQUEST => {
+                                    if exit_code == SVM_EXITCODE_SNP_EXTENDED_GUEST_REQUEST {
+                                        warn!("Fetching extended guest request is not supported");
+                                        // We don't support extended guest request, so we just write empty data.
+                                        // This matches the behavior of KVM in Linux 6.11.
+
+                                        // Read RAX & RBX from the GHCB.
+                                        let mut rax_rw_gpa_arg: mshv_read_write_gpa =
+                                            mshv_bindings::mshv_read_write_gpa {
+                                                base_gpa: ghcb_gpa + GHCB_RAX_OFFSET,
+                                                byte_count: std::mem::size_of::<u64>() as u32,
+                                                ..Default::default()
+                                            };
+                                        self.fd.gpa_read(&mut rax_rw_gpa_arg).map_err(|e| {
+                                            cpu::HypervisorCpuError::GpaRead(e.into())
+                                        })?;
+                                        let rax = u64::from_le_bytes(
+                                            <[u8; 8]>::try_from(&rax_rw_gpa_arg.data[..8]).unwrap(),
+                                        );
+                                        let mut rbx_rw_gpa_arg: mshv_read_write_gpa =
+                                            mshv_bindings::mshv_read_write_gpa {
+                                                base_gpa: ghcb_gpa + GHCB_RBX_OFFSET,
+                                                byte_count: std::mem::size_of::<u64>() as u32,
+                                                ..Default::default()
+                                            };
+                                        self.fd.gpa_read(&mut rbx_rw_gpa_arg).map_err(|e| {
+                                            cpu::HypervisorCpuError::GpaRead(e.into())
+                                        })?;
+                                        let rbx = u64::from_le_bytes(
```

Comment:
Can we just use data_npages here? Why do we need another variable if not used anywhere other than assigning this to data_npages?

Reply:
My goal was to separate the process of reading of the registers from their usage. If you feel like this is needlessly verbose, I'm also fine using `data_gpa` and `data_npages` directly.